FirebaseStorageUITests/FUIImageViewCategoryTests.m

```diff
@@ -0,0 +1,125 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import XCTest;
+
+@import FirebaseStorageUI;
+@import OCMock;
+
+#import "UIImageView+FirebaseStorage.h"
+
+@interface FUIImageViewCategoryTests : XCTestCase
+@property (nonatomic, readwrite) UIImageView *imageView;
+@property (nonatomic, readwrite) SDImageCache *cache;
+@end
+
+@implementation FUIImageViewCategoryTests
+
+- (void)setUp {
+  [super setUp];
+  self.cache = OCMClassMock([SDImageCache class]);
+  self.imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+}
+
+- (void)tearDown {
+  [super tearDown];
+}
+
+- (void)testItCreatesADownloadTaskIfCacheIsEmpty {
+  FIRStorageReference *ref = OCMClassMock([FIRStorageReference class]);
+  OCMStub([ref fullPath]).andReturn(@"path/to/image.png");
+  OCMStub([ref dataWithMaxSize:512 completion:[OCMArg any]])
+    .andReturn(OCMClassMock([FIRStorageDownloadTask class]));
+  FIRStorageDownloadTask *download = [self.imageView sd_setImageWithStorageReference:(FIRStorageReference *)ref
+                                                                        maxImageSize:512
+                                                                    placeholderImage:nil
+                                                                               cache:self.cache
+                                                                          completion:^(UIImage *image,
+                                                                                       NSError *error,
+                                                                                       SDImageCacheType cacheType,
+                                                                                       FIRStorageReference * storageRef) {
+    XCTAssert(self.imageView.image == image, @"expected download to populate image");
+    XCTAssert(error == nil, @"expected successful download to not produce an error");
+  }];
+  XCTAssert(download != nil, @"expected image view with empty cache to attempt a download");
+}
+
+- (void)testItDoesNotCreateADownloadIfImageIsCached {
+  FIRStorageReference *ref = OCMClassMock([FIRStorageReference class]);
```

Comment:
Consider pulling the mock ref up to the top and constructing it in setUp, then resetting it in tearDown